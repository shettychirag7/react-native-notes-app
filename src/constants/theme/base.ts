import {Theme} from './type';
import {fonts} from '../fonts';

const base: Theme = {
  fontFamily: {
    titleLight: fonts.montserrat.light,
    titleRegular: fonts.montserrat.semiBold,
    titleBold: fonts.montserrat.bold,
    bodyLight: fonts.montserrat.thin,
    bodyBold: fonts.montserrat.medium,
    bodyRegular: fonts.montserrat.regular,
  },
  fontSize: {
    sm: 14,
    md: 16,
    lg: 20,
    xl: 24,
    xxl: 32,
  },
  space: {
    xs: 4,
    sm: 8,
    md: 16,
    lg: 24,
    xl: 32,
    xxl: 40,
  },
  colors: {
    primary: {
      10: '#aceef5',
      20: '#70e3f0',
      30: '#00d7ea',
      40: '#00cee6',
      50: '#00c4e1',
      60: '#00b4ce',
      70: '#009fb3',
      80: '#008b9a',
      90: '#00676c',
    },
    secondary: {
      10: '#f5f5f5',
      20: '#eeeeee',
      30: '#e0e0e0',
      40: '#bebebe',
      50: '#9f9f9f',
      60: '#767676',
      70: '#626262',
      80: '#424242',
      90: '#212121',
    },
    neutral: {
      N10: '#F7F7F7',
      N20: '#E1E1E1',
      N30: '#A8A8A8',
      N40: '#757575',
      N50: '#565656',
      N60: '#3B3B3B',
      N70: '#222222',
      N80: '#121212',
      white: '#FFFFFF',
      black: '#000000',
    },
  },
};

export default base;
