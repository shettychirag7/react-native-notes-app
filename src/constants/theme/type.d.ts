export type Theme = {
  fontFamily: {
    titleLight: string;
    titleRegular: string;
    titleBold: string;
    bodyLight: string;
    bodyBold: string;
    bodyRegular: string;
  };
  fontSize: {
    sm: number;
    md: number;
    lg: number;
    xl: number;
    xxl: number;
  };
  space: {
    xs: number;
    sm: number;
    md: number;
    lg: number;
    xl: number;
    xxl: number;
  };
  colors: {
    primary: {
      10: string;
      20: string;
      30: string;
      40: string;
      50: string;
      60: string;
      70: string;
      80: string;
      90: string;
    };
    secondary: {
      10: string;
      20: string;
      30: string;
      40: string;
      50: string;
      60: string;
      70: string;
      80: string;
      90: string;
    };
    neutral: {
      N10: string;
      N20: string;
      N30: string;
      N40: string;
      N50: string;
      N60: string;
      N70: string;
      N80: string;
      white: string;
      black: string;
    };
  };
};
