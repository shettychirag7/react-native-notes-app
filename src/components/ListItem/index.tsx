import React from 'react';
import {TouchableOpacity, TouchableOpacityProps} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import Text from '../Text';
import useStylePicker from './listitem.style';

interface ListItemProps extends TouchableOpacityProps {
  name: string;
}

const ListItem = (props: ListItemProps) => {
  const {name, ...rest} = props;

  const styles = useStylePicker();

  return (
    <TouchableOpacity {...rest} style={styles.listitem}>
      <Text type="boldBody" style={styles.title}>
        {name}
      </Text>
      <Icon name="chevron-right" size={24} />
    </TouchableOpacity>
  );
};

export default ListItem;
