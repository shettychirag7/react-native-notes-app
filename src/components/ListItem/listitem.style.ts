import {StyleSheet} from 'react-native';

import {createStylePickerHook, GeneratorProps} from '../../hooks/style';

const getStyleObject = ({theme}: GeneratorProps) =>
  StyleSheet.create({
    listitem: {
      flex: 1,
      alignItems: 'center',
      backgroundColor: 'white',
      shadowColor: 'grey',
      shadowOffset: {width: 0, height: 0},
      shadowOpacity: 0.2,
      shadowRadius: theme.space.xs,
      elevation: theme.space.sm,
      flexDirection: 'row',
      justifyContent: 'space-between',

      padding: theme.space.md,
    },
    title: {
      marginVertical: theme.space.sm,
      marginHorizontal: theme.space.md,
    },
  });

const styles = createStylePickerHook(getStyleObject);

export default styles;
