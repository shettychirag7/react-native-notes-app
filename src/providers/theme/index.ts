import {createContext} from 'react';
import themes from '../../constants/theme';

export const ThemeContext = createContext({
  theme: themes.base,
});

export const ThemeProvider = ThemeContext.Provider;
