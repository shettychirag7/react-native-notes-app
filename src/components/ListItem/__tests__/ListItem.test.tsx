import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import ListItem from '..';
import {ThemeProvider} from '../../../providers/theme';
import themes from '../../../constants/theme';

describe('ListItem component test', () => {
  it('renders ListItem component', () => {
    const {toJSON} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <ListItem name="Category 1" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('attaches onPress callback', () => {
    const mockOnPress = jest.fn();
    const {getByTestId} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <ListItem name="Category 1" testID="category_1" onPress={mockOnPress} />
      </ThemeProvider>,
    );

    fireEvent.press(getByTestId('category_1'));

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('triggers onUpdate callback on text input', () => {
    const mockOnPress = jest.fn();
    const {getByTestId} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <ListItem name="Category 1" onUpdate={mockOnPress} />
      </ThemeProvider>,
    );

    getByTestId('ip_field').props.onEndEditing({
      nativeEvent: {
        text: '123',
      },
    });

    expect(mockOnPress).toHaveBeenCalledWith('123');
  });
});
