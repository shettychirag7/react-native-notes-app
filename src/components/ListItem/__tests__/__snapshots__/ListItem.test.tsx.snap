// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component test renders ListItem component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "elevation": 8,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "padding": 16,
      "shadowColor": "grey",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 16,
          "letterSpacing": 0.1,
        },
        Object {
          "marginHorizontal": 16,
          "marginVertical": 8,
        },
      ]
    }
  >
    Category 1
  </Text>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î—Œ
  </Text>
</View>
`;
